akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala

```diff
@@ -226,6 +229,14 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
     rule { oneOrMore(addr).separatedBy(listSep) ~ EOI ~> (`X-Forwarded-For`(_)) }
   }
 
+  def `x-forwarded-host` = rule {
+    (`ip-v4-address` | `ip-v6-address`) ~ EOI ~> (b ⇒ `X-Forwarded-Host`(RemoteAddress(b)))
+  }
+
+  def `x-forwarded-proto` = rule {
+    token ~ EOI ~> (`X-Forwarded-Proto`(_))
```

Comment:
Instead of `token`, this could be more strict by using zeroOMore([`scheme-char`](https://github.com/akka/akka-http/blob/82ac13a71c7342ce5887fdc9bb7a313f14b1659f/akka-http-core/src/main/scala/akka/http/impl/model/parser/CharacterClasses.scala#L55))

Reply:
Shouldn't it be exactly once?
I don't remember seeing any rule for that, hence the plain `token`

Reply:
True, I don't know if we want to accept any char or only valid schema chars. BTW, should obviously be ```oneOrMore(`schema-char`)```.